homeassistant/components/victron_gx_mqtt/entity.py

```diff
@@ -0,0 +1,159 @@
+"""Common code for Victron Venus integration."""
+
+from abc import abstractmethod
+import logging
+from typing import Any
+
+from victron_mqtt import (
+    Device as VictronVenusDevice,
+    Metric as VictronVenusMetric,
+    MetricNature,
+    MetricType,
+)
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorStateClass
+from homeassistant.const import EntityCategory, UnitOfTime
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ENTITIES_CATEGORY_DIAGNOSTIC,
+    ENTITIES_DISABLE_BY_DEFAULT,
+    ENTITY_PREFIX,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class VictronBaseEntity(Entity):
+    """Implementation of a Victron Venus base entity."""
+
+    def __init__(
+        self,
+        device: VictronVenusDevice,
+        metric: VictronVenusMetric,
+        device_info: DeviceInfo,
+        entity_platform: str,
+        simple_naming: bool,
+        installation_id: str,
+    ) -> None:
+        """Initialize the entity."""
+        self._device = device
+        self._metric = metric
+        self._device_info = device_info
+        if simple_naming:
+            entity_id = f"{entity_platform}.{ENTITY_PREFIX}_{metric.unique_id}"
+        else:
+            entity_id = f"{entity_platform}.{ENTITY_PREFIX}_{installation_id}_{metric.unique_id}"
+        self._attr_unique_id = entity_id
+        self.entity_id = entity_id
```

Comment:
Ah this is simple naming. Instead we should have HA decide on the entity_ids. As in, with `_attr_has_entity_name = True` it will be `{device_name}_{entity_name}`. And users can rename devices as they like